### ISSUE TITLE
v40.1 on macOS 13.1 BROKEN

### ISSUE THREAD

**Issue by CalidumCanis** [rmottola/Arctic-Fox#128]
I downloaded and ran v40.1 on macOS 13.1 and the whole window except the title bar is just transparent.

**Comment by rmottola** [comment]
Never tested on such a new system. 40.2 should provide some release if you can build, but no guarantee. On 13.1 you have full Firefox and SeaMonkey...

**Comment by rmottola** [comment]
how does 42.0 fare, just out of curiosity?
ON 10.11 it works very well, on 10.13 it works too, but windows cannot be moved.

**Comment by rmottola** [comment]
@CalidumCanis please try 43.0 It works fine on 10.11, on 10.13 it works but windows cannot be dragged... test and say :) Several mac fixes were imported in this release.

**Comment by rmottola** [comment]
@CalidumCanis any news? otherwise I will probably close the bug, since I cannot reproduce. There were several updates to test too.

**Comment by katze26** [comment]
Hey it's me from another account, on macOS 12.7.4 trying out 44.0 it's opening, but there's this blank popup in the middle of the screen which isn't closing, I can see the "About:" page has loaded behind. This is how it looks:
![image](https://github.com/rmottola/Arctic-Fox/assets/167756944/96ab456f-97bc-4468-878d-4e087b79764f)

**Comment by rmottola** [comment]
@katze26 no idea... 12.7.4 is so new... probably nobody ever tested ArcticFox on it... I can't support it without having it. If nobody steps up... It could be an error message which is not readable? could you try opening --jsdebug console on launch?
It could even be a compatibilty of a 10.7 build on 12.x? ever tried opening PaleMoon 27 or Firefox 45 on such recent mac?

**Comment by rmottola** [comment]
It would be interesting to try a build done on 10.13 instead of older ones, but I cannot attach it to this bug, too big (36MB vs 25MB allowed)

**Comment by rmottola** [comment]
@CalidumCanis, @katze26 would you mind testing v45? please test the classic 10.7 build but also the specific 10.13 build I did and see if there are any differences?

**Comment by rmottola** [comment]
Please let try and let me know, trying to understand if either by miracle it was fixed or building on a newer system helps. More than that I cannot to myself.

**Comment by Jazzzny** [comment]
I think this was fixed by UXP in https://repo.palemoon.org/MoonchildProductions/UXP/commit/e73a2fb0fc21de86db411a1a0fd475582455c5ac - would it be possible to cherry pick this? It seems to be a rather simple patch.

**Comment by rmottola** [comment]
@Jazzzny I am little bit sceptic because I see nothing similar in Firefox... not even esr128, I checked all esr versions, they didn't add such a workaround. Unfortunately, I don't have such a recent mac and also we don't know if esr has such behaviour.

**Comment by rmottola** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1818939

**Comment by rmottola** [comment]
@CalidumCanis @Jazzzny  are you able to test Firefox ESR versions and see which one did not work and which do? I suppose esr45 will be broken given https://bugzilla.mozilla.org/show_bug.cgi?id=1818939

esr52 ? esr60? esr68? that could help us look for a fix.

**Comment by Jazzzny** [comment]
Unfortunately there seems to be another bug that affects Firefox 57-69, where all of these versions crash for me (at least on macOS 15, arm64).

Firefox 56 and 52 ESR launch but have the invisible window issue and Firefox 70 and 78 ESR launch without the issue.

This is a very big gap between versions, maybe the crashing isn't an issue on Intel if anyone would like to test.

**Comment by rmottola** [comment]
@Jazzzny I too fear that testing on ARM chips with emulation might add some compliaction. Best would be to reporduce on Intel and check the version you cannot test. Somebody else can join?

**Comment by Jazzzny** [comment]
I think this is how SeaMonkey fixed it:
https://gitlab.com/seamonkey-project/seamonkey-2.53-mozilla/-/commit/5d6becd447783848b00b156d12cd696864ecb07c and
https://gitlab.com/seamonkey-project/seamonkey-2.53-mozilla/-/commit/114176cff80971fef7a2c74af70d8b2f31814457

**Comment by rmottola** [comment]
> I think this was fixed by UXP in https://repo.palemoon.org/MoonchildProductions/UXP/commit/e73a2fb0fc21de86db411a1a0fd475582455c5ac - would it be possible to cherry pick this? It seems to be a rather simple patch.

Did so, it is in. Doesn't cause apparent regressions.

**Comment by rmottola** [comment]
@Jazzzny @CalidumCanis @katze26  if you would mind trying out latest release...

**Comment by Jazzzny** [comment]
<img width="1470" alt="Image" src="https://github.com/user-attachments/assets/2f6a69b1-84e3-45a8-83ef-d893c64de272" />

Confirmed working on macOS 15

**Comment by rmottola** [comment]
Would you mind checking that v46.0 still works? @Jazzzny

**Comment by Jazzzny** [comment]
Yes, arcticfox-46.0.en-US.mac64-10.11.dmg is working on macOS 15 arm64.

<img width="1382" alt="Image" src="https://github.com/user-attachments/assets/51545dbd-7adc-427e-8d74-c80199d10b65" />

**Comment by rmottola** [comment]
@Jazzzny wonderful. Confirms things did not break further. Then, alas, they should even be a little better in certain minor details of shadows and colors. Asking because of #254
